File: 496-next-greater-element-i/next-greater-element-i.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        HashMap<Integer, Integer> my = new HashMap<>();
        for (int i = 0; i < nums2.length; i++) {
            my.put(nums2[i], i);
        }
        int[] ans = new int[nums1.length];
        for (int j = 0; j < nums1.length; j++) {
            boolean found = false;

            // Start searching in nums2 from the index of nums1[j]
            for (int p = my.get(nums1[j]); p < nums2.length; p++) {
                if (nums2[p] > nums1[j]) {  
                    ans[j] = nums2[p];
                    found = true;  
                    break;
                }
            }

            // If no greater element is found, set -1
            if (!found) {
                ans[j] = -1;
            }
        }

        return ans;  // Return 
    

    }
}

File: 496-next-greater-element-i/next-greater-element-i.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        my={}
        for i in range(len(nums2)):
            my[nums2[i]] = i
        stack=[]
        for k in nums1:
            for s in range(my[k],len(nums2)):
                if k < nums2[s]:
                    stack.append(nums2[s])
                    break
            else:
                stack.append(-1)
        return stack


        # ans=[]
        # for i in range(len(nums1)):
        #     for j in range(i+1,len(nums2)):
        #         if nums1[i] < nums2[j]:
        #             ans.append(nums2[j])
        #             break
        #     else:
        #         ans.append(-1)
        # print(ans)
